### ISSUE TITLE
[Feature request] On image save generate txt file with parameters

### ISSUE THREAD

**Issue by andzejsp** [comfyanonymous/ComfyUI#73]
Would be nice that for each image that gets generated you get the parameter txt file with prompt, size, seed, cfg, samplet, steps, etc.

**Comment by geeknessfr** [comment]
You can use an EXIF Tool to see parameters inside each PNG File.
With this tool you could see this line:

```
Prompt                          : {"3": {"inputs": {"seed": 648312782957426, "steps": 32, "cfg": 6.5, "sampler_name": "dpmpp_sde", "scheduler": "karras", "denoise": 1.0, "model": ["4", 0], "positive": ["6", 0], "negative": ["7", 0], "latent_image": ["5", 0]}, "class_type": "KSampler"}, "4": {"inputs": {"ckpt_name": "illuminatiDiffusionV1_v11.safetensors"}, "class_type": "CheckpointLoaderSimple"}, "5": {"inputs": {"width": 512, "height": 768, "batch_size": 1}, "class_type": "EmptyLatentImage"}, "6": {"inputs": {"text": "masterpiece best quality ...... etc....
```

You could use this one https://exiftool.org/
Download it and drag-and-drop PNG Files over the executable to see Exif infos.
But there are others tools with graphic interfaces.

Hope that helps ;)

PS: For now with Comfy when you generate many images in one batch, the Seed is not updated for each image... I'll see if there's already an issue talking about that.

**Comment by andzejsp** [comment]
> You can use an EXIF Tool to see parameters inside each PNG File. With this tool you could see this line:
> 
> ```
> Prompt                          : {"3": {"inputs": {"seed": 648312782957426, "steps": 32, "cfg": 6.5, "sampler_name": "dpmpp_sde", "scheduler": "karras", "denoise": 1.0, "model": ["4", 0], "positive": ["6", 0], "negative": ["7", 0], "latent_image": ["5", 0]}, "class_type": "KSampler"}, "4": {"inputs": {"ckpt_name": "illuminatiDiffusionV1_v11.safetensors"}, "class_type": "CheckpointLoaderSimple"}, "5": {"inputs": {"width": 512, "height": 768, "batch_size": 1}, "class_type": "EmptyLatentImage"}, "6": {"inputs": {"text": "masterpiece best quality ...... etc....
> ```
> 
> You could use this one https://exiftool.org/ Download it and drag-and-drop PNG Files over the executable to see Exif infos. But there are others tools with graphic interfaces.
> 
> Hope that helps ;)
> 
> PS: For now with Comfy when you generate many images in one batch, the Seed is not updated for each image... I'll see if there's already an issue talking about that.

while there are tools, and such its just unnecessary additional steps in order to get the seed or prompt from the image. 
1) you have to get the tool
2) you have to drag and drop the image on the tool
3) then u get this output:
![image](https://user-images.githubusercontent.com/80409979/225042650-1978d562-042c-46e5-aa45-38ad6148d9f7.png)
4) then you have to copy the text output into some sort of text editor, (another tool)
5) then you have to format the json etc, just to copy the prompt :(
![image](https://user-images.githubusercontent.com/80409979/225043469-d95decd3-99a6-47d2-9301-265deea3eda0.png)

i know beggars cant be choosers, therefore its a feature request. :)

**Comment by geeknessfr** [comment]
Wow... it's just a temporary solution lol
I didn't say youre not right to ask that, I just said you COULD use another way temporarly.... Just trying to help you ^^

**Comment by andzejsp** [comment]
No offense, im just showing you how uncomfortable it is to get the prompts, if its not thought about at this time when the repo is in its infancy. 

Another solution then would be drag n drop the rendered img on the canvas in the repo UI and it would create nodes with all the values, not sure tho how would it work tho. 

Sometimes you just want to get the prompts :)

**Comment by angrysky56** [comment]
Comfy needs a module when you drag to it you can load the image into the workflow, not make another workflow. The module should be able to read png info into a text window and output the img to the workflow at least, ideally have options to load available params automatically from png- such as cfg, steps, samplers, upscalers, loras, model. 

The img loader would be better if you could select a whole file instead of just one img too.

**Comment by morphles** [comment]
What? Why? First "cfg, steps" irelevant, this is A11 thinking, comfy flows can be arbitrary complex, and such simple numbers do not say much. But if you just load image, you get full workflow, so what is the problem? Also what is "whole file instead of one image"?!

**Comment by angrysky56** [comment]
Yeah, I don't care about loading params much either, just some like to I know. Load a file of images so you don't have to load them one by one- have it check a folder even i.e. output, so new images are automatically loaded in and you can just click over. I don't want a different workflow, I want the one I am using to edit images. 

I want to be able to drag and drop into the load image area because that is an important thing to me for ease of editing new images.
![2023-05-06 (1)](https://user-images.githubusercontent.com/113643118/236638468-b037e9bc-55d5-4d43-90e9-c41d4af8d3c8.png)

![2023-05-06](https://user-images.githubusercontent.com/113643118/236638417-e5f0ba6f-f42e-45c8-96b6-a715f4e622f7.png)

**Comment by LIQUIDMIND111** [comment]
ANY UPDATES on how to save parameters on a TXT file so we can SEE what we use to generate an IMAGE?

**Comment by angrysky56** [comment]
I found a node to do that- not sure where maybe searge or the one button prompt.
![image](https://github.com/comfyanonymous/ComfyUI/assets/113643118/43dc23a6-3e19-4681-af0d-fdeabcadca42)

**Comment by Tekaiguy** [comment]
I believe when you drag an image onto a node that has settings (ksampler, cliptextencode, ...) the node will adopt the settings that were used to create the image. Haven't tested in a while but will confirm later.

**Comment by christian-byrne** [comment]
Yes, the image itself has all the settings/parameters used to generate it in the metadata -- same with videos, audios, 3d models. You can drop the generated outputs back into the app to reload the workflow with the parameters primed.

It's unlikely for the text file alternative to be added as a feature, but I think you can find a custom node for it. Alternatively, all the scripts used to extract the workflow/parameters from outputs are open-sourced and found here: https://github.com/Comfy-Org/ComfyUI_frontend/tree/main/src/scripts/metadata.

You can also just use `exiftools` on commandline and look at WORKFLOW tag.
